### ISSUE TITLE
Fails to build with Carthage

### ISSUE THREAD

**Issue by rpowelll** [ReSwift/ReSwift-Router#20]
It seems ReSwift Router fails to build properly through Carthage given an empty directory with the following `Cartfile`:

```
github "ReSwift/ReSwift" ~> 1.0
github "ReSwift/ReSwift-Router"
```

Here's the build output (ignore the Swift 2.2 warnings):

```
$ carthage update --platform ios
*** Fetching ReSwift-Router
*** Fetching ReSwift
*** Checking out ReSwift at "1.0.0"
*** Checking out ReSwift-Router at "0.2.6"
*** Fetching Swift-Flow
*** xcodebuild output can be found in /var/folders/mn/xfk73_v56ns232tl5s448zdm0000gn/T/carthage-xcodebuild.6HC7Uc.log
*** Building scheme "ReSwift-iOS" in ReSwift.xcodeproj
*** Building scheme "ReSwiftRouter-iOS" in ReSwiftRouter.xcodeproj
** BUILD FAILED **


The following build commands failed:
	CompileSwift normal arm64 /Users/rpowell/Code/ReSwiftBugs/Carthage/Checkouts/ReSwift-Router/ReSwiftRouter/NavigationReducer.swift
	CompileSwift normal arm64 /Users/rpowell/Code/ReSwiftBugs/Carthage/Checkouts/ReSwift-Router/ReSwiftRouter/NavigationState.swift
	CompileSwift normal arm64 /Users/rpowell/Code/ReSwiftBugs/Carthage/Checkouts/ReSwift-Router/ReSwiftRouter/Router.swift
	CompileSwift normal arm64 /Users/rpowell/Code/ReSwiftBugs/Carthage/Checkouts/ReSwift-Router/ReSwiftRouter/NavigationActions.swift
	CompileSwift normal arm64 /Users/rpowell/Code/ReSwiftBugs/Carthage/Checkouts/ReSwift-Router/ReSwiftRouter/Routable.swift
	CompileSwiftSources normal arm64 com.apple.xcode.tools.swift.compiler
(6 failures)
/Users/rpowell/Code/ReSwiftBugs/Carthage/Checkouts/ReSwift-Router/ReSwiftRouter/NavigationReducer.swift:34:26: warning: 'var' parameters are deprecated and will be removed in Swift 3
/Users/rpowell/Code/ReSwiftBugs/Carthage/Checkouts/ReSwift-Router/ReSwiftRouter/NavigationReducer.swift:42:9: warning: 'var' parameters are deprecated and will be removed in Swift 3
/Users/rpowell/Code/ReSwiftBugs/Carthage/Checkouts/ReSwift-Router/ReSwiftRouter/NavigationReducer.swift:9:8: error: no such module 'ReSwift'
/Users/rpowell/Code/ReSwiftBugs/Carthage/Checkouts/ReSwift-Router/ReSwiftRouter/NavigationReducer.swift:34:26: warning: 'var' parameters are deprecated and will be removed in Swift 3
/Users/rpowell/Code/ReSwiftBugs/Carthage/Checkouts/ReSwift-Router/ReSwiftRouter/NavigationReducer.swift:42:9: warning: 'var' parameters are deprecated and will be removed in Swift 3
/Users/rpowell/Code/ReSwiftBugs/Carthage/Checkouts/ReSwift-Router/ReSwiftRouter/NavigationReducer.swift:9:8: error: no such module 'ReSwift'
/Users/rpowell/Code/ReSwiftBugs/Carthage/Checkouts/ReSwift-Router/ReSwiftRouter/NavigationReducer.swift:34:26: warning: 'var' parameters are deprecated and will be removed in Swift 3
/Users/rpowell/Code/ReSwiftBugs/Carthage/Checkouts/ReSwift-Router/ReSwiftRouter/NavigationReducer.swift:42:9: warning: 'var' parameters are deprecated and will be removed in Swift 3
/Users/rpowell/Code/ReSwiftBugs/Carthage/Checkouts/ReSwift-Router/ReSwiftRouter/NavigationReducer.swift:9:8: error: no such module 'ReSwift'
/Users/rpowell/Code/ReSwiftBugs/Carthage/Checkouts/ReSwift-Router/ReSwiftRouter/NavigationReducer.swift:34:26: warning: 'var' parameters are deprecated and will be removed in Swift 3
/Users/rpowell/Code/ReSwiftBugs/Carthage/Checkouts/ReSwift-Router/ReSwiftRouter/NavigationReducer.swift:42:9: warning: 'var' parameters are deprecated and will be removed in Swift 3
/Users/rpowell/Code/ReSwiftBugs/Carthage/Checkouts/ReSwift-Router/ReSwiftRouter/NavigationReducer.swift:9:8: error: no such module 'ReSwift'
/Users/rpowell/Code/ReSwiftBugs/Carthage/Checkouts/ReSwift-Router/ReSwiftRouter/NavigationReducer.swift:34:26: warning: 'var' parameters are deprecated and will be removed in Swift 3
/Users/rpowell/Code/ReSwiftBugs/Carthage/Checkouts/ReSwift-Router/ReSwiftRouter/NavigationReducer.swift:42:9: warning: 'var' parameters are deprecated and will be removed in Swift 3
/Users/rpowell/Code/ReSwiftBugs/Carthage/Checkouts/ReSwift-Router/ReSwiftRouter/NavigationReducer.swift:9:8: error: no such module 'ReSwift'
A shell task failed with exit code 65:
** BUILD FAILED **
```

**Comment by Ben-G** [comment]
@rpowelll Thanks for reporting this! I can reproduce this issue locally and will look into it as soon as possible!

**Comment by Ben-G** [comment]
With what I've investigated so far it seems that ReSwift-Router was relying on having the carthage dependencies checked into git. I've changed that with the latest release. 

This seems to break ReSwiftRouter with Carthage, because Carthage does not check out the entire subproject, instead it only provides the built binary in the build folder. 

This means I should reference the built framework in ReSwiftRouter's Xcode project to fix this issue.
